psyneulink/core/compositions/parameterestimationcomposition.py

```diff
@@ -815,9 +903,37 @@ def run(self, *args, **kwargs):
 
         # Capture the input passed to run and pass it on to the OCM
         assert self.controller is not None
-        self.controller.set_pec_inputs_cache(
-            kwargs.get("inputs", None if not args else args[0])
-        )
+
+        # Get the inputs
+        inputs = kwargs.get("inputs", None if not args else args[0])
+
+        # Since we are passing fitting\optimization parameters as inputs we need add them to the inputs
+        if inputs:
+
+            # Don't check inputs if we are within a call to evaluate_agent_rep, the inputs have already been checked and
+            # cached on the PEC controller.
+            if ContextFlags.PROCESSING not in context.flags:
+                self.controller.check_pec_inputs(inputs)
+
+            # Copy the inputs so we don't modify the original dict, note, we can't copy the keys because they
+            # are object\mechanisms that are in the underlying composition.
+            inputs = {k: v.copy() for k, v in inputs.items()}
```

Comment:
Yeah, the values, at least in the case that they are numpy arrays already, will be modified in place I believe when construct the parameter values for the control signals. See this [method](https://github.com/PrincetonUniversity/PsyNeuLink/blob/efa28752d0db751d28edcd068c91b8f8f18a7733/psyneulink/core/compositions/parameterestimationcomposition.py#L1208). They will probably get copied in this code (depending on the branch they go down), but I thought it was safer to copy them explicitly here just in case one of the branches leads to modifying them in place.